Apeer/fiji_scripts/fijipytools.py

```diff
@@ -633,27 +626,26 @@ class ThresholdTools:
 
     @staticmethod
     def apply_autothreshold(hist, method='Otsu'):
-
-        if method == 'Otsu':
-            lowthresh = Auto_Threshold.Otsu(hist)
-        if method == 'Triangle':
-            lowthresh = Auto_Threshold.Triangle(hist)
-        if method == 'IJDefault':
-            lowthresh = Auto_Threshold.IJDefault(hist)
-        if method == 'Huang':
-            lowthresh = Auto_Threshold.Huang(hist)
-        if method == 'MaxEntropy':
-            lowthresh = Auto_Threshold.MaxEntropy(hist)
-        if method == 'Mean':
-            lowthresh = Auto_Threshold.Mean(hist)
-        if method == 'Shanbhag':
-            lowthresh = Auto_Threshold.Shanbhag(hist)
-        if method == 'Yen':
-            lowthresh = Auto_Threshold.Yen(hist)
-        if method == 'Li':
-            lowthresh = Auto_Threshold.Li(hist)
-
-        return lowthresh
+        # I think the use of a dict here makes it a little easier to see what's going on, and avoids repetition
+        method_dict = {'Otsu': Auto_Threshold.Otsu,
+                      'Triangle': Auto_Threshold.Triangle,
+                      'IJDefault': Auto_Threshold.IJDefault,
+                      'Huang': Auto_Threshold.Huang,
+                      'MaxEntropy': Auto_Threshold.MaxEntropy,
+                      'Mean': Auto_Threshold.Mean,
+                      'Shanbhag': Auto_Threshold.Shanbhag,
+                      'Yen': Auto_Threshold.Yen,
+                      'Li': Auto_Threshold.Li
+                      }
+        if method in method_dict:
+            method_func = method_dict[method]
+            lowthresh = method_func(hist)
+            return lowthresh
+        else:
+            print(f"Method passed not found: {method}")
+            # or print("Method passed not found: {}".format(method) or print("Method passed not found: %s" % method)
+            # whichever works best for you
+            return None
```

Comment:
Same as above. Either throw or return None. In any case you can omit the if-else statement.

Reply:
Same as above